numpy/core/src/multiarray/shape.c

```diff
@@ -160,11 +160,18 @@ PyArray_Resize(PyArrayObject *self, PyArray_Dims *newshape, int refcheck,
     }
 
     /* make new_strides variable */
-    _array_fill_strides(
-        new_strides, new_dimensions, new_nd, PyArray_DESCR(self)->elsize,
-        PyArray_FLAGS(self), &(((PyArrayObject_fields *)self)->flags));
-    memmove(PyArray_DIMS(self), new_dimensions, new_nd*sizeof(npy_intp));
-    memmove(PyArray_STRIDES(self), new_strides, new_nd*sizeof(npy_intp));
+    if (new_nd > 0) {
```

Comment:
it is not necessary if the dimensions do not change, it could be moved above, needs more indentation but is probably nicer structure